tldw_Server_API/app/core/DB_Management/chacha/note_store.py

```
@@ -0,0 +1,843 @@
+            "ORDER BY last_modified DESC "
+            "LIMIT ? OFFSET ?"
+        )
+        params.extend([limit, offset])
+        cursor = self._db.execute_query(query, tuple(params))
+        return [dict(row) for row in cursor.fetchall()]
+
+    def list_deleted_notes(self, limit: int = 100, offset: int = 0) -> list[dict[str, Any]]:
+        """List only soft-deleted notes (trash)."""
+        return self.list_notes(limit=limit, offset=offset, only_deleted=True)
+
+    def get_notes_batch(self, note_ids: list[str], include_deleted: bool = True) -> list[dict[str, Any]]:
```

Ruff (0.15.11)

[warning] Boolean-typed positional argument in function definition

(FBT001)